internal/hcs/vmcompute.go

```diff
@@ -0,0 +1,549 @@
+package hcs
+
+import (
+	gcontext "context"
+	"syscall"
+
+	"github.com/Microsoft/hcsshim/internal/logfields"
+	"github.com/Microsoft/hcsshim/internal/timeout"
+	"github.com/sirupsen/logrus"
+	"go.opencensus.io/trace"
+)
+
+const (
+	stuckSyscallWarning = "Syscall did not complete within operation timeout. This may indicate a platform issue. If it appears to be making no forward progress, obtain the stacks and see if there is a syscall stuck in the platform API for a significant length of time."
+)
+
+func hcsEnumerateComputeSystemsContext(ctx gcontext.Context, query string, computeSystems **uint16, result **uint16) (hr error) {
+	ctx, span := trace.StartSpan(ctx, "HcsEnumerateComputeSystems")
+	defer span.End()
```

Comment:
@kevpar / @jstarks - I cant decide on this one. On the one hand I really want logs to have the error from the platform here. This allows us to have span duration for all vmcompute syscalls as well as results. On the other hand this part of the stack does not have the context to know if the error was expected or not. So part of me says "dont log errors, return them". If the caller did not expect an error then it should log it.

What do you think? Should I add the `defer func() { oc.SetSpanStatus(span, hr) }()` to this?

Reply:
I think it's okay to fail the span here. I don't think it will muddy our analytics, because if we just want to see errors surfaced to CRI, we can just filter to only failures on the CRI entry-point spans.

Reply:
Ok I will add it then